<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FormGen48</class>
 <widget class="QWidget" name="FormGen48">
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="0" column="0">
    <widget class="QWidget" name="widget" native="true">
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item row="0" column="0" colspan="2">
       <widget class="QTabWidget" name="tabWidget">
        <property name="currentIndex">
         <number>3</number>
        </property>
        <widget class="QWidget" name="tabAuto">
         <attribute name="title">
          <string>Auto/None</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="labelAutoDesc">
            <property name="text">
             <string>Automatically select suitable options based on conditions.</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelAuto">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabQuadF">
         <attribute name="title">
          <string>Quad-feature</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="labelSalt">
            <property name="toolTip">
             <string>Value (such as a structure salt) added to the 48-bit seeds</string>
            </property>
            <property name="text">
             <string>Add structure salt:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineSalt">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="labelLow20">
            <property name="toolTip">
             <string>Selects a set of values for the lower 20-bits of the seed, corresponding to quad-feature constellations
Applies only to feature-structures of region-size = 32 and chunk-gap = 8, in particular swamp-huts</string>
            </property>
            <property name="text">
             <string>Restrict lower bits:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="StyledComboBox" name="comboLow20">
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToMinimumContentsLength</enum>
            </property>
            <item>
             <property name="text">
              <string>Quad-hut (ideal)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Quad-hut (classic)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Quad-hut (normal)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Quad-hut (barely)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Quad-feature (ideal, custom salt)</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabQuadM">
         <attribute name="title">
          <string>Quad-monument</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="labelQM">
            <property name="toolTip">
             <string>Number of planar spawning spaces inside monuments that are within 128 blocks of a center position
(i.e. this is a measure quality, out of 4*58*58)</string>
            </property>
            <property name="text">
             <string>Min. monument area:</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tabList">
         <attribute name="title">
          <string>Seed list</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_7">
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="labelPath">
            <property name="toolTip">
             <string>Use a custom seed list as source for 48-bit seed candidates</string>
            </property>
            <property name="text">
             <string>Load 48-bit seed list:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="lineList48">
            <property name="font">
             <font>
              <family>Monospace</family>
             </font>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="buttonBrowse">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="labelListSalt">
            <property name="toolTip">
             <string>Value (such as a structure salt) added to the 48-bit seeds</string>
            </property>
            <property name="text">
             <string>Add structure salt:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QLineEdit" name="lineListSalt">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string>0</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="1">
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::MinimumExpanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="0" column="2">
          <widget class="QRadioButton" name="radioAuto">
           <property name="text">
            <string>Auto</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QRadioButton" name="radioManual">
           <property name="text">
            <string>Manual</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="labelTranspose">
           <property name="text">
            <string>Transpose by regions (multiplied by ×512):</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="labelX1">
          <property name="toolTip">
           <string>Lower bound (inclusive)</string>
          </property>
          <property name="text">
           <string>X&lt;sub&gt;1&lt;/sub&gt;:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="CoordEdit" name="lineEditX1">
          <property name="toolTip">
           <string>Lower bound (inclusive)</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelZ1">
          <property name="toolTip">
           <string>Lower bound (inclusive)</string>
          </property>
          <property name="text">
           <string>Z&lt;sub&gt;1&lt;/sub&gt;:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="CoordEdit" name="lineEditZ1">
          <property name="toolTip">
           <string>Lower bound (inclusive)</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>    </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelX2">
          <property name="toolTip">
           <string>Upper bound (inclusive)</string>
          </property>
          <property name="text">
           <string>X&lt;sub&gt;2&lt;/sub&gt;:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="CoordEdit" name="lineEditX2">
          <property name="toolTip">
           <string>Upper bound (inclusive)</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="labelZ2">
          <property name="toolTip">
           <string>Upper bound (inclusive)</string>
          </property>
          <property name="text">
           <string>Z&lt;sub&gt;2&lt;/sub&gt;:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="CoordEdit" name="lineEditZ2">
          <property name="toolTip">
           <string>Upper bound (inclusive)</string>
          </property>
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelSpace">
        <property name="text">
         <string>Seeds in search space:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="labelCount">
        <property name="text">
         <string>0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>0</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CoordEdit</class>
   <extends>QLineEdit</extends>
   <header>src/conditiondialog.h</header>
  </customwidget>
  <customwidget>
   <class>StyledComboBox</class>
   <extends>QComboBox</extends>
   <header>src/widgets.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
